api/src/routes/edit.ts

```diff
@@ -319,6 +341,72 @@ edit.put("/object/:pid/state", requireToken, pidSanitizer, bodyParser.text(), as
 });
 
 const pidAndParentPidSanitizer = sanitizeParameters({ pid: pidSanitizeRegEx, parentPid: pidSanitizeRegEx });
+edit.put(
+    "/object/:pid/parent/:parentPid",
+    requireToken,
+    pidAndParentPidSanitizer,
+    bodyParser.text(),
+    async function (req, res) {
+        try {
+            const pid = req.params.pid;
+            const parent = req.params.parentPid;
+            const pos = parseInt(req.body);
+
+            // Validate the input
+            if (pid == parent) {
+                res.status(400).send("Object cannot be its own parent.");
+                return;
+            }
+            const parentData = await FedoraDataCollector.getInstance().getHierarchy(parent);
+            if (parentData.getAllParents().includes(pid)) {
+                res.status(400).send("Object cannot be its own grandparent.");
+                return;
+            }
+            if (!parentData.models.includes("vudl-system:CollectionModel")) {
+                res.status(400).send("Illegal parent " + parent + "; not a collection!");
+                return;
+            }
+
+            // If we got this far, we can safely update things
+            const fedoraObject = FedoraObject.build(pid);
+            await fedoraObject.addParentRelationship(parent);
+            if (parentData.sortOn === "custom") {
+                await fedoraObject.addSequenceRelationship(parent, pos);
+            }
+            res.status(200).send("ok");
+        } catch (error) {
+            console.error(error);
+            res.status(500).send(error.message);
+        }
+    }
+);
+edit.delete("/object/:pid/parent/:parentPid", requireToken, pidAndParentPidSanitizer, async function (req, res) {
+    try {
+        const pid = req.params.pid;
+        const parent = req.params.parentPid;
+        const fedora = Fedora.getInstance();
+
+        // Validate the input
+        const fedoraData = await FedoraDataCollector.getInstance().getHierarchy(pid);
+        const legalParent = fedoraData.parents.reduce((previous, current) => {
```

Comment:
Good point, though actually I think it's Array.find() since we want to return the matching element rather than just a boolean status; see 8369602bf43f504c2d28902a82cd4256480c4a2d.